sqlmesh/core/dialect.py

```diff
@@ -795,38 +795,40 @@ def pandas_to_sql(
     )
 
 
-def set_default_schema_and_catalog(
+def set_default_catalog(
     table: str | exp.Table,
-    default_schema: t.Optional[str],
     default_catalog: t.Optional[str],
 ) -> exp.Table:
     table = exp.to_table(table)
 
-    if default_schema and not table.db:
-        table.set("db", exp.to_identifier(default_schema))
-    if default_catalog and not table.catalog:
-        table.set("catalog", exp.to_identifier(default_catalog))
+    if default_catalog and not table.catalog and table.db:
+        table.set("catalog", exp.parse_identifier(default_catalog))
```

Comment:
minor issue. this should probably take the project dialect because if someone passes in a catalog with back ticks, it's not gonna work. this assumes double quotes.

if this is too hard, put this on the todo.